examples/hello-world/index.js

```diff
@@ -3,7 +3,11 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-const application = (module.exports = require('./dist'));
+const nodeMajorVersion = +process.versions.node.split('.')[0];
+const dist = nodeMajorVersion >= 10 ? './dist10' : './dist8';
+const application = require(dist);
+
+module.exports = application;
```

Comment:
Does it make sense to extract this logic to a separate module such as `@loopback/dist` as follows?
```js
const path = require('path');
module.exports = function requireDistribution(projectRootDir) {
  const nodeMajorVersion = +process.versions.node.split('.')[0];
  const dist = nodeMajorVersion >= 10 ? './dist10' : './dist8';
  return require(path.join(projectRootDir, dist));
}
```

Reply:
A better place for it might be `@loopback/build/dist` or `dist-select`?

Reply:
`@loopback/build` is a dev dependency for most of our packages. We need `dist` to be a regular dep.

Reply:
I see your point.

My counter-argument is package/dependency size. In order to share these three lines of code, LB4 users would have to download yet another package from npmjs.org.

I don't have a strong opinion on this though. Considering that both you @raymondfeng and @virkt25 are in favor of this change, I'll follow your lead here.